wolfssl/wolfcrypt/types.h

```diff
@@ -1377,7 +1377,7 @@ typedef struct w64wrapper {
         typedef unsigned int  THREAD_RETURN;
         typedef size_t        THREAD_TYPE;
         #define WOLFSSL_THREAD
-    #elif (defined(_POSIX_THREADS) || defined(HAVE_PTHREAD))
+    #elif defined(WOLFSSL_PTHREADS)
```

Comment:
We've had a lot of issues with this type of auto-detection. What is the reasoning to not use `WOLFSSL_PTHREADS`?

Reply:
@julek-wolfssl Could you clarify? `WOLFSSL_PTHREADS` IS being used

Reply:
Lol. Hopefully :)